homeassistant/components/surepetcare/sensor.py

```diff
@@ -93,3 +113,22 @@ def _update_attr(self) -> None:
         else:
             self._attr_extra_state_attributes = {}
         _LOGGER.debug("%s -> state: %s", self.name, state)
+
+
+class Flap(SurePetcareSensor):
+    """Sure Petcare Flap."""
+
+    def __init__(self, _id: int, coordinator: DataUpdateCoordinator) -> None:
+        """Initialize a Sure Petcare Flap sensor."""
+        super().__init__(_id, coordinator)
+
+        self._attr_name = f"{self._device_name} Flap"
+        self._attr_unique_id = f"{self._device_id}-flap"
+
+    @callback
+    def _update_attr(self, surepy_entity) -> None:
+        """Update the state and attributes."""
+        status = surepy_entity.raw_data()["status"]
+        self._attr_native_value = LockState(status["locking"]["mode"]).name.capitalize()
+        self._attr_extra_state_attributes = {"learn_mode": bool(status["learn_mode"])}
```

Comment:
What does learn mode represent?

Reply:
The is for a cat flap, where only registered cats (with chips) are allowed to enter.
Learn mode is what you enable to register a new cat. 
Not sure how useful it is, but it was in the old integration.

Reply:
Maybe it should be a separate binary sensor?

Reply:
fyi: this mode is rarely used by a typical user, basically just a single time for each cat on the initial setup. So I think an own sensor for this isn't really needed.

Reply:
Then it should be an entity that is disabled by default.